engine/internal/processor/processor.go

```diff
@@ -174,32 +159,48 @@ func (p *P) NeedLeaderElection() bool {
 	return true
 }
 
-func (p *P) run(ctx context.Context) error {
+// Start runs the processor.
+//
+// TODO(kenji): Gracefully handle an error from the server.
+func (p *P) Start(ctx context.Context) error {
+	log := p.logger.WithValues("engineID", p.engineID)
+	log.Info("Starting processor")
+	ctx = ctrl.LoggerInto(ctx, log)
 	ctx = auth.AppendWorkerAuthorization(ctx)
-	stream, err := p.client.ProcessTasks(ctx)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		_ = stream.CloseSend()
-	}()
-
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
 
-	errCh := make(chan error, 2)
-	go func() {
-		if err := p.sendEngineStatusPeriodically(ctx, stream); err != nil {
-			errCh <- fmt.Errorf("send engine status: %s", err)
+	run := func() error {
+		streamCtx, streamCancel := context.WithCancel(context.Background())
```

Comment:
Out of of scope of this PR, but can we make `inference-manager-engine` stop receiving new tasks when grace period starts?

Reply:
got it!